src/fmt.zig

```diff
@@ -133,27 +135,76 @@ fn formatIdent(fmt: *Formatter, ident: TokenIdx, qualifier: ?TokenIdx) void {
     fmt.pushTokenText(ident);
 }
 
-fn formatExpr(fmt: *Formatter, ei: ExprIdx) void {
+const Braces = enum {
+    round,
+    square,
+    curly,
+    bar,
+
+    fn start(b: Braces) u8 {
+        return switch (b) {
+            .round => '(',
+            .square => '[',
+            .curly => '{',
+            .bar => '|',
+        };
+    }
+
+    fn end(b: Braces) u8 {
+        return switch (b) {
+            .round => ')',
+            .square => ']',
+            .curly => '}',
+            .bar => '|',
+        };
+    }
+};
+
+fn formatCollection(fmt: *Formatter, region: IR.Region, braces: Braces, comptime T: type, items: []T, formatter: fn (*Formatter, T) IR.Region) void {
+    const multiline = fmt.ast.regionIsMultiline(region);
+    fmt.push(braces.start());
+    if (multiline) {
+        fmt.curr_indent += 1;
+    }
+    var i: usize = 0;
+    for (items) |item| {
+        if (multiline) {
+            fmt.newline();
+            fmt.pushIndent();
+        }
+        const arg_region = formatter(fmt, item);
+        if (!multiline and i < (items.len - 1)) {
+            fmt.pushAll(", ");
+        }
+        if (multiline) {
+            fmt.push(',');
+            fmt.flushCommentsAfter(arg_region);
+        }
+        i += 1;
+    }
+    if (multiline) {
+        fmt.curr_indent -= 1;
+        fmt.newline();
+        fmt.pushIndent();
+    }
+    fmt.push(braces.end());
+}
+
+fn formatExpr(fmt: *Formatter, ei: ExprIdx) IR.Region {
```

Comment:
I noticed the returned region is almost always ignored. What is it actually needed/used for?

Reply:
It's used by formatCollection to flush comments and newlines (normalized) between the items

Reply:
See fmt.zig:212 above